telegram-bot-api/Client.cpp

```diff
@@ -7704,6 +8050,232 @@ td::Status Client::process_get_memory_stats_query(PromisedQueryPtr &query) {
+                 std::make_unique<TdOnReturnChatCallback>(this, std::move(query)));
+  } else {
+    fail_query(400, "Bad request: Please specify chat_id or invite_link", std::move(query));
+  }
+
+  return Status::OK();
+}
+
+td::Status Client::process_add_chat_member_query(PromisedQueryPtr &query) {
+  CHECK_IS_USER();
+  auto chat_id = query->arg("chat_id");
+  TRY_RESULT(user_id, get_user_id(query.get()))
+
+  check_chat(chat_id, AccessRights::Write, std::move(query),
+             [this, user_id](int64 chat_id, PromisedQueryPtr query) mutable {
+               auto chat = get_chat(chat_id);
+               if (chat->type == ChatInfo::Type::Supergroup) {
+                 std::vector<td::int32> user_ids{user_id};
+                 send_request(make_object<td_api::addChatMembers>(chat_id, std::move(user_ids)),
+                              std::make_unique<TdOnOkQueryCallback>(std::move(query)));
+               } else if (chat->type == ChatInfo::Type::Group) {
+                 send_request(make_object<td_api::addChatMember>(chat_id, user_id, 0),
+                              std::make_unique<TdOnOkQueryCallback>(std::move(query)));
+               }
+             });
+  return Status::OK();
+}
+
+td::Status Client::process_report_chat_query(PromisedQueryPtr &query) {
+  CHECK_IS_USER();
+  auto chat_id = query->arg("chat_id");
+  TRY_RESULT(reason, get_report_reason(query.get()));
+  TRY_RESULT(message_ids, get_int_array_arg<td::int64>(query.get(), "message_ids", true));
+
+  check_chat(chat_id, AccessRights::Read, std::move(query),
+             [this, reason = std::move(reason), message_ids = std::move(message_ids)](int64 chat_id,
+                                                                                      PromisedQueryPtr query) mutable {
+               send_request(make_object<td_api::reportChat>(chat_id, std::move(reason), std::move(message_ids)),
+                            std::make_unique<TdOnOkQueryCallback>(std::move(query)));
+             });
+  return Status::OK();
+}
+
+td::Status Client::process_create_chat_query(PromisedQueryPtr &query) {
+  CHECK_IS_USER();
+  auto chat_type = query->arg("type");
+  auto title = query->arg("title");
+
+  if (chat_type == "supergroup") {
+    send_request(make_object<td_api::createNewSupergroupChat>(title.str(), false, "", nullptr),
+                 std::make_unique<TdOnReturnChatCallback>(this, std::move(query)));
+  } else if (chat_type == "channel") {
+    send_request(make_object<td_api::createNewSupergroupChat>(title.str(), true, "", nullptr),
+                 std::make_unique<TdOnReturnChatCallback>(this, std::move(query)));
+  } else if (chat_type == "group") {
+    TRY_RESULT(initial_members, get_int_array_arg<td::int32>(query.get(), "user_ids"))
+    send_request(make_object<td_api::createNewBasicGroupChat>(std::move(initial_members), title.str()),
+                 std::make_unique<TdOnReturnChatCallback>(this, std::move(query)));
+  }
+  return Status::OK();
```

Comment:
Handle the case when chat_type is something else

Reply:
It raises an error now, done